bin/plugin/commands/readme.js

```diff
@@ -48,20 +55,53 @@ exports.handler = async ( opt ) => {
 		owner: config.githubRepositoryOwner,
 		repo: config.githubRepositoryName,
 		milestone: opt.milestone,
+		path: opt.path,
 		token: opt.token,
 	} );
 };
 
+/**
+ * Detects the path to the readme.txt file to update based on the milestone title.
+ *
+ * @param {string} milestone Milestone title.
+ *
+ * @return {string} Detected readme.txt path.
+ */
+function detectReadmePath( milestone ) {
+	const slug = milestone.match( /^([a-z0-9-]+) / );
+	if ( ! slug ) {
+		throw new Error(
+			`The ${ milestone } milestone does not start with a valid plugin slug.`
+		);
+	}
+
+	if ( 'performance-lab' === slug[ 1 ] ) {
+		return 'readme.txt';
+	}
+
```

Comment:
It could be useful to also check if the path `plugins/${ slug[ 1 ] }/readme.txt` exists as an additional check.

```suggestion

	if ( ! fs.existsSync( `plugins/${ slug[ 1 ] }` ) ) {
		throw new Error( `Unknown plugin with slug '${ slug[ 1 ] }'` );
	}

```

Otherwise, If I add a typo like:

```bash
npm run readme -- -m "speculation-rulez 1.1.0"
```

Then I get an error:

> Error: Cannot find milestone by title: speculation-rulez 1.1.0

But this is the same error you'd get if the version had a typo.

Just an additional safeguard.

Reply:
Great idea, fixed in https://github.com/WordPress/performance/pull/1075/commits/fcd1af2288118b9c4211b2b7602d6ee48251348d.